### ISSUE TITLE
Adding Eigen::Transform

### ISSUE THREAD

**Issue by ahundt** [jrl-umi3218/Eigen3ToPython#13]
I'm thinkikng of adding Eigen::Transform https://github.com/RLovelett/eigen/blob/master/Eigen/src/Geometry/Transform.h to this, could you advise me on where I'll need to make changes?

**Comment by gergondet** [comment]
You'd need to do 2 things:
- `c_eigen.pxd` to declare `Eigen::Transform<T, dim. mode>` and the operations you want to access and typedef for `Affine3d` (you could also just declare the `Affine3d` class and the functions you want but doing so with `Transform` is the better solution)
- Write code in `generate_eigen_pyx.py` that writes the binding for a specific instance of Transform (e.g. `Affine3d`) and append that output to the generated `eigen.pyx` (around L. 668)
- Add the declaration to `eigen.pxd` (also generated, around L. 686 of `generate_eigen_pyx.py`)

Note: for the declaration of `mode` you'll want to do something similar to the way `dynamic`, `one`, `two`, `three`, `four` and `six` are defined but using Eigen's nameds instead, e.g.

```python
ctypedef int Affine "Affine"
```

I think @haudren tried this or something related in the past as well (but I might be remembering wrong)

**Comment by haudren** [comment]
@gergondet : Yes I tried to use those things in the past, but it was in a bit more complex situation.

Here you are in a typical "bind an enum" case. Thus you can use any of the following things:
```python
cdef extern int one "eins", two "zwei"
cdef extern float three "drei"

cdef struct spam "SPAM":
    int i "eye"

cdef enum surprise "inquisition":
    first "alpha"
    second "beta" = 3
```
(taken from [here](http://docs.cython.org/en/latest/src/userguide/external_C_code.html#styles-of-struct-union-and-enum-declaration) )

I would recommend going with the third style, as it neatly maps the Python from the C++ (and nobody expects the Spanish inquisition).
